### ISSUE TITLE
Hints to set up a dev environment for the project (IntelliJ, Eclipse)

### ISSUE THREAD

**Issue by hjohn** [javafxports/openjdk-jfx#187]
The build instructions are clear enough, however getting the project imported in either IntelliJ or Eclipse seems to be hard (I didn't manage it).  In both IntelliJ and Eclipse it's mainly the tests that are not compiled correctly because of missing stuff on the classpath (`org.junit` is missing), but there are other issues as well, like `com.sun.javafx.logging.PlatformLogger` not being found.

Can some simple instructions be added how to properly import this project into one of these IDE's?

**Comment by nlisker** [comment]
I don't know about IntelliJ, but Eclipse has a pending issue to fix it's project files: https://bugs.openjdk.java.net/browse/JDK-8209015

You can go to the webrev there, download and apply the patch, and configuration should work.

As for instructions, yes, they are outdated.

**Comment by nlisker** [comment]
@hjohn I update the instructions on the Wiki (WIP). See if it's better now.

**Comment by elect86** [comment]
I see both tons of errors in Idea and Netbeans

Do you, guys, use an IDE to develop? If yes, can you specify your step to get a working development environment? 

Also operating system and jdk (everything seems to be configured for 8, but on the wiki the hint is 11)

**Comment by elect86** [comment]
Ok, [found](https://wiki.openjdk.java.net/display/OpenJFX/Using+an+IDE)

**Comment by nlisker** [comment]
Yes, 2018-09. I wrote the instructions on Windows, but I don't think it should matter.
They are here: https://wiki.openjdk.java.net/display/OpenJFX/Using+an+IDE#UsinganIDE-UsingEclipse

The images are a remnant from an older version of the page (couldn't update them). If you need Java+JavaFX 8 you can look at the page history. Admittedly, this page should have been cloned and maintained as 2 versions.

**Comment by elect86** [comment]
Ok, so, I re-cloned this repo in my home, not in Eclipse Workspace.

After checking `Add project to working sets` shall I also create a new one? If yes, which type?

**Comment by nlisker** [comment]
A working set is optional and you can create one at any time later. The type would be Java.

**Comment by elect86** [comment]
Ok, so, I set as installed JRE the openjdk 11.01 and as Compiler Level, I put 10 (no 11 yet)

When I `Apply and Close`, the workspace builds, but at the end I get this:

![pic](https://i.imgur.com/Zql6BL8.png)

**Comment by nlisker** [comment]
Compiler version 11 is available with the patch from http://download.eclipse.org/eclipse/updates/4.9-P-builds/. I added this to the instructions page.

All of the projects that belong to SceneBuilder (under /rt/apps/scenebuilder) can be removed from the workspace since they are no longer maintained and planned to be removed (https://bugs.openjdk.java.net/browse/JDK-8198961).

After you set the compiler level to 11 `base` should compile. What error is it giving you?

**Comment by elect86** [comment]
Ok, the patched enabled level 11.

But I keep getting errors, unresolved imports specifically:

![pic](https://i.imgur.com/1urLgyA.png)

**Comment by nlisker** [comment]
What is the error on `base` project?

**Comment by elect86** [comment]
I have an exclamation mark on that, but I cant find any more info about it.

Opening some java files, such as `src/main/java/module-info.java`, everythin is red:
> Syntax error on token(s), misplaced constructor(s)

![pic](https://i.imgur.com/1jeoTvc.png)

`com.sun.javafx.beans.IDProperty.java`:
> Syntax error, annotations are only available if source level is 1.5 or higher

I double checked, under Window -> Preferences -> Java -> Compiler, level is 11

But if I `Properties` on the project itself, such as `base`:

![pic](https://i.imgur.com/MYwCtc2.png)

**Comment by elect86** [comment]
I checked `Enable project specific settings` and selected level 11

Then Project -> Clean

Now `module-info.java` is not completely red, only the project name modules:

![pic](https://i.imgur.com/z79htcD.png)

> the package javafx.beans does not exist or is empty

**Comment by elect86** [comment]
I guess the safest and easiest method is to delete your openjfx project, clone it fresh and annotate down all the step you are gonna do in order to solve all the errors and get a working stable developing environment

**Comment by nlisker** [comment]
Your `module-info.java` should be directly under the source folder and not the default package. I don't know why you have it this way. It also looks like you have a configuration error on `base` because of the red `!`. In the problem view try to show only problems on what you select (the downward arrow to the top right) and see what it reports specifically on the project level.

**Comment by elect86** [comment]
![pic](https://i.imgur.com/lBUxaZP.png)

I'll clone it once again.

Did you clone from here or from openjdk.net?

**Comment by nlisker** [comment]
I cloned from the mercurial repo as described in the build instructions, but they are (or should be) the same.

Now I see the problem. For some reason you are missing a dependency. Open the build path configuration of the project. It should look like this:

![image](https://user-images.githubusercontent.com/37422899/48666928-00ab6900-ead4-11e8-9940-95225f827ac0.png)

The first entry is what the build complains about. If it is not declared there then you didn't clone properly or something like that because those dependencies are declared in the .classpath files provided with the repo. You classpath file for `base` should look like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src/main/java"/>
	<classpathentry kind="src" path="build/gensrc/java"/>
	<classpathentry kind="src" output="testbin" path="src/shims/java">
		<attributes>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="testbin" path="src/test/java">
		<attributes>
			<attribute name="test" value="true"/>
			<attribute name="optional" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4">
		<attributes>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER">
		<attributes>
			<attribute name="module" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="output" path="bin"/>
</classpath>
```

If that folder is declared and you see a red error mark on it then it's probably missing and you should try to execute `gradle` to build the repo once without Eclipse.

**Comment by elect86** [comment]
Ok, I'll take note.

One question: is it important to build it at least once from console before proceding with the IDE?

**Comment by nlisker** [comment]
You need to execute the gradle build once. It creates build resources that are required by the IDE. The Eclipse instructions show how you can do it with Buildship, so you don't have to use the console, but this you would want to test that building works when you initially clone the repo it was assumed that you built at least once already.

**Comment by elect86** [comment]
Also, might be a problem openjdx installed under `C:\Program Files\Java\jdk-11.0.1` instead of `C:\Program Files\Java\jdk-11` as the wiki says?

**Comment by nlisker** [comment]
If you configured the environment variables to the JDK path, no matter what it is, then it doesn't matter. If the system can't find the JDK then nothing will work (gradle build, Eclipse, or any Java application). The build instructions say that they assume a certain path and that "you should adjust these as needed for your system."

**Comment by elect86** [comment]
Ok, I re-cloned, build it first and then opened in Eclipse.

Now `module-info.java` is indeed under the root and `base` doesnt show errrors.

However, others still do:

![pic](https://i.imgur.com/9Are4GL.png)

**Comment by nlisker** [comment]
As I [said](https://github.com/javafxports/openjdk-jfx/issues/187#issuecomment-439638044), you should remove all the SceneBuilder projects from the workspace (look at the nested projects too). `buildSrc` is not a Java project so you don't need it.

As for the errors, I don't know what project they belong to. All the modules seem to be fine.

**Comment by elect86** [comment]
I recall I did removed 3 projects under apps/scenebuilder at importing the project

**Comment by nlisker** [comment]
Look under scenebuilder\samples.

**Comment by elect86** [comment]
Ok, thanks

Ps: which is the simplest sample to test graphic backed?

**Comment by nlisker** [comment]
What exactly do you want to test? You can create a new project in Eclipse and set it up as in the instructions.

**Comment by kevinrushforth** [comment]
> As I said, you should remove all the SceneBuilder projects from the workspace

On a related note, I plan to send out a patch for [JDK-8198961](https://bugs.openjdk.java.net/browse/JDK-8198961) in the next few weeks to remove the obsolete and unmaintained SceneBuilder sources from the JavaFX repo. SceneBuilder is being maintained by Gluon in the [gluonhq/scenebuilder](https://github.com/gluonhq/scenebuilder) project on GitHub.

**Comment by elect86** [comment]
I'd like to test the graphic gl es2 backed.

Anyway, I re-cloned the project once again (for testing win10 instructions) and now I get this:

![pic](https://i.imgur.com/PTLaWvj.png)

**Comment by nlisker** [comment]
Didn't you already have all the modules build? I don't know where the error of the package comes from. Is it empty or non-existent?

**Comment by elect86** [comment]
yeah, but I wanted/needed to test if the issues on win10 I [had](https://github.com/javafxports/openjdk-jfx/issues/288#issuecomment-439955639), had to do with the gradle-daemon

`javafx.beans` isnt empty: I have an `IDProperty.java` under it

**Comment by nlisker** [comment]
It's really hard to debug like that. I don't see how the instructions could lead to this situation and I don't see the errors on `base`. If you just clone and import `base` you should at worst get 1 error about `genSrc` being missing.
